### PR TITLE
Add `paywallViewControllerRequestedDismissal`

### PR DESCRIPTION
**Pull request by vegaro**
Please read https://github.com/RevenueCat/react-native-purchases/pull/913

Depends on https://github.com/RevenueCat/purchases-ios/pull/3738

### CODE REVIEW

**Review by tonidero — COMMENTED** [review]
Changes LGTM! Lmk when this is ready for review

**Comment by vegaro** [comment]
@tonidero I am waiting for https://github.com/RevenueCat/purchases-ios/pull/3738, which needs to be released before this can be merged

**Review by tonidero — APPROVED** [review]

File: ios/PurchasesHybridCommon/PurchasesHybridCommonUI/PaywallProxy.swift
Comment (tonidero) [inline ios/PurchasesHybridCommon/PurchasesHybridCommonUI/PaywallProxy.swift:41]: We need to set this delegate for the footer methods as well. Otherwise it won't be called after a purchase completes

File: ios/PurchasesHybridCommon/PurchasesHybridCommonUI/PaywallProxy.swift
Comment (tonidero) [inline ios/PurchasesHybridCommon/PurchasesHybridCommonUI/PaywallProxy.swift:41]: We need to update the name of this parameter and add it to the footer methods... Maybe we can even extract to a function to avoid repeating the code?

Comment (tonidero) [reply ios/PurchasesHybridCommon/PurchasesHybridCommonUI/PaywallProxy.swift:41]: Though thinking about it, we might need to make `self` weak here I think?

**Review by tonidero — APPROVED** [review]
LGTM!

File: ios/PurchasesHybridCommon/PurchasesHybridCommonUI/PaywallProxy.swift
Comment (tonidero) [inline ios/PurchasesHybridCommon/PurchasesHybridCommonUI/PaywallProxy.swift:29]: As mentioned, I think we might want to weakify `self` here?

Comment (vegaro) [reply ios/PurchasesHybridCommon/PurchasesHybridCommonUI/PaywallProxy.swift:29]: I think so yes

File: ios/PurchasesHybridCommon/PurchasesHybridCommonUI/PaywallProxy.swift
Comment (tonidero) [inline ios/PurchasesHybridCommon/PurchasesHybridCommonUI/PaywallProxy.swift:189]: Hmm thinking about it, we could also hold just the delegate (weakly) instead of self... But this should be fine.

**Review by tonidero — APPROVED** [review]

File: ios/PurchasesHybridCommon/PurchasesHybridCommonUI/PaywallProxy.swift
Comment (vegaro) [reply ios/PurchasesHybridCommon/PurchasesHybridCommonUI/PaywallProxy.swift:189]: actually the delegate is already weak

    public weak var delegate: PaywallViewControllerDelegateWrapper?

Comment (vegaro) [reply ios/PurchasesHybridCommon/PurchasesHybridCommonUI/PaywallProxy.swift:189]: but yes, let's leave it like this, I tested it just in case and it works fine.

**Pull request merged**
